File: hrms/hr/page/employee_basic_view/employee_basic_view.py
import frappe

@frappe.whitelist()
def get_employee_info():
    user = frappe.session.user

    # Fetching employee details based on the logged-in user
    employee = frappe.get_value("Employee", {"user_id": user}, ["employee_name", "name", "department", "designation"], as_dict=True)

    if not employee:
        return {"employee_name": "N/A", "employee_id": "N/A", "department": "N/A", "designation": "N/A"}

    return {
        "employee_name": employee.employee_name,
        "employee_id": employee.name,
        "department": employee.department,
        "designation": employee.designation
    }

File: hrms/hr/page/employee_basic_view/employee_basic_view.js
frappe.pages['employee_basic_view'].on_page_load = function(wrapper) {
    frappe.call({
        method: "frappe.client.get_value",
        args: {
            doctype: "Employee",
            filters: {
                user_id: frappe.session.user
            },
            fieldname: "name"
        },
        callback: function(r) {
            if (r && r.message && r.message.name) {
                const employee_id = r.message.name;
                // Redirecting to the workspace/route with a tab anchor
                frappe.set_route("app", "employee", employee_id);
                // Optional: delay before applying the tab hash
                setTimeout(() => {
                    window.location.hash = "tab=connections";
                }, 100);
            } else {
                frappe.msgprint("Employee record not found for the current user.");
            }
        }
    });
};
